jwala-services/src/main/java/com/cerner/jwala/service/jvm/impl/JvmServiceImpl.java

```diff
@@ -115,11 +120,18 @@ public JvmServiceImpl(final JvmPersistenceService jvmPersistenceService,
         this.binaryDistributionService = binaryDistributionService;
         this.binaryDistributionLockManager = binaryDistributionLockManager;
         this.historyFacadeService = historyFacadeService;
+        this.webServerPersistenceService = webServerPersistenceService;
         this.fileUtility = fileUtility;
     }
 
 
     protected Jvm createJvm(final CreateJvmRequest aCreateJvmRequest) {
+        List<WebServer> webServerList = webServerPersistenceService.getWebServers();
+        for (WebServer webserver : webServerList) {
+            if (aCreateJvmRequest.getJvmName().equals(webserver.getName())) {
+                throw new EntityExistsException("Webserver already exists with this name "+ aCreateJvmRequest.getJvmName());

```

Comment:
In addition I'm not in favor of using EntityExistException because at first glance of an exception, it doesn't tell where the exception occurred. I prefer using JvmStateServiceException (or JvmServiceException if this method is transferred there).